services/storage/tests/unit/test_modules_celery.py

```diff
@@ -150,25 +150,19 @@ async def test_submitting_task_with_failure_results_with_error(
     ):
         with attempt:
             raw_result = await celery_client.get_task_result(task_context, task_uuid)
-            result = TypeAdapter(TaskError).validate_python(raw_result)
-            assert isinstance(result, TaskError)
+            assert isinstance(raw_result, Exception)
```

Comment:
is this now a Exception type only? not sure that is a good idea.
How do you now differentiate the exceptions? by reading strings? or is there something more elaborate? if yes could you please enhance the test? thanks